### PR TITLE
Integration testing cleanup

### PR DESCRIPTION
**Pull request by pyrodogg**
The goal is to keep the framework for integration testing but not require screeps-server-mockup by default. Instead, to instruct users how to install the dependency themselves. 

- Adds instructions to install `screeps-server-mockup` and adding test scripts
- Removes `test-integration` script
- Removes integration from `test` script

### CODE REVIEW

**Comment by brisberg** [comment]
Should we remove the files under `test/integration` and make those optional too?

Right now they don't even compile without the `screeps-server-mockup` dependency.

**Comment by djD-REK** [comment]
Looks ready to merge to me :)

@brisberg I think the test files can stay, since you won't be running them without following the steps in the docs...

**Comment by djD-REK** [comment]
Alternatively, what would you think of just making the screeps-server-mockup dependency required for this package? Too much? 🔢

File: docs/in-depth/testing.md [hunk 1]
```diff
@@ -42,13 +42,33 @@ magnitude, it is recommended to prefer unit tests wherever possible.
 
 ## Integration Testing
 
+### Installing Screeps Server Mockup
+
+Before starting to use integration testing, you must install [screeps-server-mockup](https://github.com/screepers/screeps-server-mockup) to your project.
```
Comment (djD-REK) [inline docs/in-depth/testing.md:47]: Maybe add an example here?
```
# Using yarn:
yarn add screeps-server-mockup
# Using npm
npm install screeps-server-mockup
```

File: docs/in-depth/testing.md [hunk 2]
```diff
@@ -42,13 +42,33 @@ magnitude, it is recommended to prefer unit tests wherever possible.
...
+Now you can run integration tests by using the `test-integration` script or run both unit and integration tests using the `test` script.
+
+### Integration Testing with Screeps Server Mockup
+
 Integration testing is for code that depends heavily on having a full game
 environment. Integration tests are completely representative of the real game
 (in fact they run with an actual Screeps server). This comes at the cost of
 performance and very involved setup when creating specific scenarios.
 
 Server testing support is implmented via
```
Comment (djD-REK) [inline docs/in-depth/testing.md]: typo: implemented

File: docs/in-depth/testing.md (hunk 1 above)
Comment (pyrodogg) [reply docs/in-depth/testing.md:47]: Added

File: docs/in-depth/testing.md (hunk 2 above)
Comment (pyrodogg) [reply docs/in-depth/testing.md]: Thanks, fixed.

**Comment by pyrodogg** [comment]
@djD-REK Thanks for bringing this up again. I stalled when considering how much further it could go and forgot that the basic `test` script is currently broken on master.

**Pull request merged**